src/components/work/ContactHostFormModal/ContactHostFormModal.tsx

```diff
@@ -0,0 +1,162 @@
+import * as React from 'react';
+import { Formik, Field, Form as FormikForm } from 'formik';
+import * as Yup from 'yup';
+import { compose, graphql } from 'react-apollo';
+
+import { Booking } from 'networking/bookings';
+import { CONTACT_USER, ContactUserField, User } from 'networking/users';
+import { Button, Form, FormGroup, Label, FormFeedback, Input, ModalFooter, ModalBody, Modal, ModalHeader } from 'reactstrap';
+import Textarea from 'components/shared/Textarea';
+import { TextareaEvent } from 'components/shared/Textarea/Textarea';
+import Loading from 'components/shared/loading/Loading';
+
+interface Props {
+  contactUser: (input: ContactUserInput) => Promise<EmailResponse>;
+  booking: Booking;
+  handleModalAction: () => void;
+}
+
+interface FormValues {
+  [name: string]: string;
+}
+
+interface ContactUserInput {
+  bookingId?: string;
+  listingId?: string;
+  message: string;
+  recipientId: string;
+  subject: string;
+}
+
+interface EmailResponse {
+  bookingId: string;
+  listingId: string;
+  message: string;
+  recipient: User;
+  subject: string;
+}
+
+const defaultValues: FormValues = {
+  [ContactUserField.SUBJECT]: '',
+  [ContactUserField.MESSAGE]: '',
+};
+
+const ContactHostSchema = Yup.object({
+  subject: Yup.string().required('Please fill out the subject field.'),
+  message: Yup.string().required('Please fill out the message field.'),
+});
+
+function ContactHostForm({ booking, contactUser, handleModalAction }: Props) {
```

Comment:
Oh yay, this will come in handy for the Contact Host button on the listing page

Reply:
🎉